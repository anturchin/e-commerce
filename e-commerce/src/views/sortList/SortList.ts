import { View } from '../View';
import { InputSort } from './inputSort/InputSort';
import { FormElemetSorting } from './type';

import './SortList.scss';

export class FormSort extends View {
    private inputSort: InputSort | null = null;

    constructor() {
        super({ tag: 'form', classNames: ['form-sort'] });
        this.setupForm();
    }

    public getFormElements(): FormElemetSorting | null {
        if (this.inputSort) {
            return {
                inputSort: this.inputSort,
            };
        }
        return null;
    }

    private setupForm(): void {
        this.inputSort = new InputSort({
            label: 'Sort',
            required: false,
            options: ['By popularity', 'By discount size', 'Ascending price', 'Descending price'],
        });
        this.viewHtmlElement.addInnerElement(this.inputSort.getElement());
    }
}
