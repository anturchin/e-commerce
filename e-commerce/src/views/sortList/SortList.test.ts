import { describe, expect, test } from '@jest/globals';
import { FormSort } from './SortList';

describe('SortList', () => {
    test('should match', () => {
        const formSort = new FormSort();
        const renderedFormSort = formSort.getElement();
        expect(renderedFormSort.outerHTML).toMatchSnapshot();
    });
});
