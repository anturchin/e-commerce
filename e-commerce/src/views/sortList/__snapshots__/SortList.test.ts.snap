// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortList should match 1`] = `"<form class="form-sort"><select class="input" name="Sorting"><option value="By popularity">By popularity</option><option value="By discount size">By discount size</option><option value="Ascending price">Ascending price</option><option value="Descending price">Descending price</option></select></form>"`;
